log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java

```diff
@@ -666,11 +665,12 @@ protected List<Node> processSelect(final Node selectNode, final PluginType<?> ty
     }
 
     protected void doConfigure() {
-        instanceFactory.registerBinding(Binding.from(StringValueResolver.KEY).toInstance(configurationStrSubstitutor));
+        instanceFactory.registerBinding(StringValueResolver.KEY, this::getConfigurationStrSubstitutor);
         processConditionals(rootNode);
         preConfigure(rootNode);
         configurationScheduler.start();
-        if (rootNode.hasChildren() && rootNode.getChildren().get(0).getName().equalsIgnoreCase("Properties")) {
+        if (rootNode.hasChildren()
+                && "Properties".equalsIgnoreCase(rootNode.getChildren().get(0).getName())) {
```

Comment:
Various flipping around is from nullability analysis from some time where `Node::getName` is technically nullable.